airflow/models/dag.py

```diff
@@ -2207,6 +2207,9 @@ def partial_subset(
                 also_include.extend(t.get_flat_relatives(upstream=False))
             if include_upstream:
                 also_include.extend(t.get_flat_relatives(upstream=True))
+            else:
+                upstream_setup_tasks = t.get_flat_relatives(upstream=True, setup_only=True)
+                also_include.extend(upstream_setup_tasks)
```

Comment:
which task do you think it's missing? 
i did add tests so pretty sure it's working properly.

i bet that the confusing bit here is that, there is logic, in `get_flat_relatives` such that, whenever a setup is returned, its teardown (if one exists) is returned

so, when you're going upstream, you don't need to do anything different.

but when you're going downstream, you need to add upstream setups (and their towndows)

so, with downstream, the teardowns are added on line 2207, since they are always downstream.  

and then on 2211, it's just saying "if downstream and not upstream" then we must _add_ the upstream setups.

now, you could argue that, it's not obvious from the signature / name of get_flat_relatives that when setups are returned that their teardowns are also returned.   and if that's your thought i welcome any suggestion.

Reply:
I think I didn't understand the code well. Your explanation makes sense. I was thinking that if `include_upstream` is selected by the user then the else part will not run

Reply:
yeah get_flat_relatives was updated so that.... whenever a setup is returned, it always adds the teardown.
now, this assumes that the only usage of it is clearing behavior.  i should probably verify that.